dotnetv3/Glue/Actions/GlueWrapper.cs

```diff
@@ -0,0 +1,361 @@
+        {
+            var databaseName = response.Crawler.DatabaseName;
+            Console.WriteLine($"{crawlerName} has the database {databaseName}");
+            return response.Crawler;
+        }
+
+        Console.WriteLine($"No information regarding {crawlerName} could be found.");
+        return null;
+    }
+
+    // snippet-end:[Glue.dotnetv3.GetCrawler]
+
+    // snippet-start:[Glue.dotnetv3.GetCrawlerState]
+    /// <summary>
+    /// Get information about the state of an AWS Glue crawler.
+    /// </summary>
+    /// <param name="crawlerName">The name of the crawler.</param>
+    /// <returns>A value describing the state of the crawler.</returns>
+    public async Task<CrawlerState> GetCrawlerStateAsync(string crawlerName)
+    {
+        var response = await _amazonGlue.GetCrawlerAsync(
+            new GetCrawlerRequest { Name = crawlerName });
+        return response.Crawler.State;
+    }
+
+    // snippet-end:[Glue.dotnetv3.GetCrawlerState]
+
+    // snippet-start:[Glue.dotnetv3.GetDatabase]
+    /// <summary>
+    /// Get information about an AWS Glue database.
+    /// </summary>
+    /// <param name="dbName">The name of the database.</param>
+    /// <returns>A Database object containing information about the database.</returns>
+    public async Task<Database> GetDatabaseAsync(string dbName)
+    {
+        var databasesRequest = new GetDatabaseRequest
+        {
+            Name = dbName,
+        };
+
+        var response = await _amazonGlue.GetDatabaseAsync(databasesRequest);
+        return response.Database;
+    }
+
+    // snippet-end:[Glue.dotnetv3.GetDatabase]
+
+    // snippet-start:[Glue.dotnetv3.GetJobRun]
+    /// <summary>
+    /// Get information about a specific AWS Glue job run.
+    /// </summary>
+    /// <param name="jobName">The name of the job.</param>
+    /// <param name="jobRunId">The Id of the job run.</param>
+    /// <returns>A JobRun object with information about the job run.</returns>
+    public async Task<JobRun> GetJobRunAsync(string jobName, string jobRunId)
+    {
+        var response = await _amazonGlue.GetJobRunAsync(new GetJobRunRequest { JobName = jobName, RunId = jobRunId });
+        return response.JobRun;
+    }
+
+    // snippet-end:[Glue.dotnetv3.GetJobRun]
+
+    // snippet-start:[Glue.dotnetv3.GetJobRuns]
+    /// <summary>
+    /// Get information about all AWS Glue runs of a specific job.
+    /// </summary>
+    /// <param name="jobName">The name of the job.</param>
+    /// <returns>A list of JobRun objects.</returns>
+    public async Task<List<JobRun>> GetJobRunsAsync(string jobName)
+    {
+        var jobRuns = new List<JobRun>();
+
+        var request = new GetJobRunsRequest
+        {
+            JobName = jobName,
+        };
+
+        do
+        {
+            var response = await _amazonGlue.GetJobRunsAsync(request);
+            jobRuns.AddRange(response.JobRuns);
+            if (response.NextToken is not null)
```

Comment:
Looks like there is a paginator for GetJobRuns, is there a reason not to use it?

Reply:
Changed to use the paginator.